### PR TITLE
changes for checking queuesize cronjob

### PR DESCRIPTION
**Pull request by avi-hmcts**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
     chmod 777 /var/spool/exim && \
     ln -sf /dev/stdout /var/log/exim/mainlog && \
     ln -sf /dev/stderr /var/log/exim/panic && \
     ln -sf /dev/stderr /var/log/exim/reject && \
-    chmod 0755 /usr/sbin/exim; \
+    chmod 0755 /usr/sbin/exim && \
+    chmod 0755 /usr/sbin/crond; \
 else \
-    apk --no-cache add exim logrotate && \
+    apk --no-cache add exim logrotate bash && \
```
Comment (luigibk) [inline Dockerfile]: why do you need bash?

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
     sed -i 's/28/7/' /etc/logrotate.d/exim && \
     mkdir /var/spool/exim && \
     chmod 777 /var/spool/exim && \
-    chmod 0755 /usr/sbin/exim; \
+    chmod 0755 /usr/sbin/exim && \
+    chmod 0755 /usr/sbin/crond; \
```
Comment (luigibk) [inline Dockerfile]: Why is crond added everywhere with no oprtion to exclude it?

File: Dockerfile [hunk 3]
```diff
@@ -35,4 +35,7 @@ ENV LOCAL_DOMAINS=@ \
     SMTP_USERDOMAIN= \
     SMTP_USERNAME=
 
-CMD ["exim", "-bdf", "-q15m"]
+ADD crontab.txt /crontab.txt
+ADD mailqueue.sh /mailqueue.sh
+
+CMD ["exim -bdf -q15m" ; "crond -l 8 -f"]
```
Comment (luigibk) [inline Dockerfile]: I don't like the idea of running 2 processes in the same container.

**Comment by luigibk** [comment]
I am not convinced about running a new process in the same container. Why don't we try to scrape the exim log files (which are configurable too) from a sidecar container using an exporter such as mtail?

**Comment by avi-hmcts** [comment]
> I am not convinced about running a new process in the same container. Why don't we try to scrape the exim log files (which are configurable too) from a sidecar container using an exporter such as mtail?

As per our discussion on slack addressed your comments and updated the PR accordingly.

File: helm/exim/values.yaml [hunk 4]
```diff
@@ -80,3 +80,10 @@ fluentbit:
     repository: fluent/fluent-bit
     tag: 1.7.2
     pullPolicy: IfNotPresent
+
+mtail:
+  enabled: true
```
Comment (luigibk) [inline helm/exim/values.yaml]: enabled should be false

**Review by luigibk — APPROVED** [review]

**Pull request merged**